gips/data/prism/prism.py

```diff
@@ -282,59 +287,50 @@ def get_bil_file(d, a):
-                    VerboseOut('Using default lag of {} days.'.format(lag), 2)
+                    utils.verbose_out('Using default lag of {} days.'.format(lag), 2)
 
                 if os.path.exists(fname):
                     os.remove(fname)
 
-                inv = self.inventory(
-                    dates='{},{}'.format(
-                        datetime.strftime(
-                            self.date - timedelta(days=lag - 1), '%Y-%m-%d',
-                        ),
-                        datetime.strftime(self.date, '%Y-%m-%d'),
+                date_spec = '{},{}'.format(
+                    datetime.strftime(
+                        self.date - timedelta(days=lag - 1), '%Y-%m-%d',
                     ),
+                    datetime.strftime(self.date, '%Y-%m-%d'),
+                )
+                inv = self.inventory(
+                    dates=date_spec,
                     products=['ppt'],
                 )
                 inv.process()
-                inv = self.inventory(
-                    dates='{},{}'.format(
-                        datetime.strftime(
-                            self.date - timedelta(days=lag - 1), '%Y-%m-%d',
-                        ),
-                        datetime.strftime(self.date, '%Y-%m-%d'),
-                    ),
+                inv = self.inventory(  # because DataInventory object doesn't update
+                    dates=date_spec,
                     products=['ppt'],
                 )
                 if len(inv.data) < lag:
-                    raise Exception(
-                        key + ': requires {} preceding days ppt assets.'
-                        .format(lag) +
-                        '\nOnly {} days found.'.format(len(inv.data))
+                    utils.verbose_out(
+                        '{}: requires {} preceding days ppt ({} found).'
+                        .format(key, lag, len(inv.data)),
+                        3,
                     )
+                    continue
                 imgs = []
                 for tileobj in inv.data.values():
                     datobj = tileobj.tiles.values()[0]
-                    imgs.append(datobj.open_assets('ppt'))
+                    imgs.append(GeoImage(get_bil_vsifile(datobj, '_ppt')))
                 oimg = GeoImage(fname, imgs[0])
                 for chunk in oimg.Chunks():
                     oarr = oimg[0].Read(chunk) * 0.0
                     for img in imgs:
-                        print('mean value of image: {}'.format(mean(img[0].Read())))
+                        utils.verbose_out(
+                            'mean value of image: {}'
+                            .format(mean(img[0].Read())),
+                            3,
+                        )
                         oarr += img[0].Read(chunk)
                     oimg[0].Write(oarr, chunk)
                 oimg.Process()
-                #oimg = None
-                print('nih')
+                oimg = None
```

Comment:
optional:  This looks like an optimization, maybe to let the garbage collector know it can deallocate `oimg`?  Might be worth it to add a comment to that effect.